manifests/init.pp

```diff
@@ -5,17 +5,20 @@
 # === Parameters
 #  [*package_ensure*]           openssl package ensure
 #  [*ca_certificates_ensure*]   ca-certificates package ensure
+#  [*libssl_dev_ensure*]        libssl-dev ensure
 #
 # === Example
 #
 #   class { '::openssl':
 #     package_ensure         => latest,
 #     ca_certificates_ensure => latest,
+#     libssl_dev_ensure      => latest,
 #   }
 #
 class openssl (
   $package_ensure         = present,
   $ca_certificates_ensure = present,
+  $libssl_dev_ensure      = present,
```

Comment:
I'm not sure how many people upstream would ever need this or whether this would be something the maintainer would accept.